packages/mdc-textfield/mdc-textfield.scss

```diff
@@ -123,10 +124,197 @@ $mdc-textfield-disabled-border-on-dark: rgba(white, .3);
+    left: 0;
+    // TODO ripple positioning logic?
+    width: 100%;
+    height: 100%;
+    border-radius: 50%;
+    background-color: inherit;
+    content: "";
+    opacity: 0;
+    pointer-events: none;
+  }
+
+  // NOTE: For some reasons, stylelint complains that the two patterns below don't follow BEM.
+  // However, it seems to emit the wrong selector for that pattern. Thus, we disable it above where
+  // we would normally disable it (&.mdc-textfield--disabled) as a workaround.
+
+  // stylelint-disable plugin/selector-bem-pattern
+  // TODO: Refactor this out once #194 lands
+  $opacity-duration: 83ms;
+
+  &::before {
+    transition: opacity 17ms linear, transform 0ms $opacity-duration+2 linear;
+  }
+
+  &:hover {
+    &::before {
+      transition:
+        opacity $opacity-duration linear 17ms,
+        transform 250ms $mdc-animation-linear-out-slow-in-timing-function 17ms;
+      opacity: 1;
+    }
+  }
+
+  .mdc-textfield__input {
+    align-self: flex-end;
+    height: 100%;
+    box-sizing: border-box;
+    padding: 20px 16px 0;
+  }
+
+  .mdc-textfield__label {
+    @include mdc-rtl-reflexive-position(left, 16px);
+
+    position: absolute;
+    bottom: 20px;
+    color: rgba(black, .6);
+    pointer-events: none;
+    // Force the label into its own layer to prevent to prevent visible layer promotion adjustments
+    // when the ripple is activated behind it.
+    will-change: transform;
+
+    @include mdc-theme-dark(".mdc-textfield") {
+      @include mdc-theme-prop(color, text-secondary-on-dark);
+    }
+
+    &--float-above {
+      transform: translateY(-50%) scale(.75, .75);
+    }
+  }
+
+  &.mdc-ripple-upgraded {
+    @include mdc-ripple-base;
+    // TODO: Add ::before as ripple bg once #194 lands
+    @include mdc-ripple-fg((pseudo: "::after", opacity: .04));
+
+    &::before {
+      top: calc(50% - var(--mdc-ripple-fg-size, 100%) / 2);
+      left: calc(50% - var(--mdc-ripple-fg-size, 100%) / 2);
+      width: var(--mdc-ripple-fg-size, 100%);
+      height: var(--mdc-ripple-fg-size, 100%);
+      transform: scale(.8);
+    }
+
+    &:hover::before {
+      transform: scale(calc(var(--mdc-ripple-fg-scale, 1) + .1));
+    }
+
+    @include mdc-theme-dark(".mdc-textfield", true) {
+      @include mdc-ripple-fg((pseudo: "::after", base-color: white, opacity: .05));
+    }
+  }
+
+  &.mdc-textfield--disabled {
+    border-bottom: none;
+    background-color: rgba(black, .02);
+    color: rgba(black, .38);
+
+    @include mdc-theme-dark(".mdc-textfield", true) {
+      border-bottom: none;
+      background-color: rgba(white, .05);
+    }
+
+    .mdc-textfield__label {
+      bottom: 20px;
+    }
+
+    .mdc-textfield__input {
+      padding-bottom: 0;
+    }
+  }
+
+  // TODO: Move into general .mdc-textfield decl and replace ::after textfield styles with this
+  // element as part of #675
+  .mdc-textfield__bottom-line {
+    position: absolute;
+    bottom: 0;
+    left: 0;
+    width: 100%;
+    height: 1px;
```

Comment:
Yeah so this is a change that hasn't yet made it into spec. When we implement #675 it'll elevate to 2dp on hover, and stay at 2dp on focus.